pulumi/infra/test_network.py

```diff
@@ -0,0 +1,67 @@
+import pulumi
+import sys
+import unittest
+from unittest.mock import MagicMock, patch
+
+
+class MyMocks(pulumi.runtime.Mocks):
+    def new_resource(self, args: pulumi.runtime.MockResourceArgs):
+        outputs = args.inputs
+        return [args.name + '_id', outputs]
+
+    def call(self, args: pulumi.runtime.MockCallArgs):
+        if args.token == "aws:index/getAvailabilityZones:getAvailabilityZones":
+            return {
+                "names": ["a", "b", "c"],
+                "state": "available"
+            }
+        # This is necessary since infra.config calls this
+        if args.token == "aws:index/getCallerIdentity:getCallerIdentity":
+            return {
+                "account_id": "000000000000"
+            }
+        return {}
+
+pulumi.runtime.set_mocks(MyMocks())
+
+sys.modules['infra.network.infra.config'] = MagicMock()
```

Comment:
Is "infra.network.infra.config" correct? That's not an existing module 🤔

Reply:
I've removed this since we don't really need to patch the DEPLOYMENT_NAME. 
That said, I'll note that when I tried to use infra.config by itself I ran into errors. It seems like you need to use infra.network in front so that pytest knows that infra.config is being called by infra.network. Seems weird though, so there's probably (hopefully?) a better way to do this. Or even better would be to get rid of the side effects so things could be imported without needing mocks